apollo-federation/src/composition/satisfiability/satisfiability_error.rs

```diff
@@ -0,0 +1,466 @@
+                        "Int" => ast::Value::Int(0.into()),
+                        "Float" => ast::Value::Float((3.1).into()),
+                        "Boolean" => ast::Value::Boolean(true),
+                        "String" => ast::Value::String("A string value".to_string()),
+                        // Users probably expect a particular format of ID at any particular place,
+                        // but we have zero info on the context, so we just throw a string that
+                        // hopefully make things clear.
+                        "ID" => ast::Value::String("<any id>".to_string()),
+                        // It's a custom scalar, but we don't know anything about that scalar so
+                        // providing some random string. This will technically probably not be a
+                        // valid value for that scalar, but hopefully that won't be enough to throw
+                        // users off.
+                        _ => ast::Value::String("<some value>".to_string()),
+                    }
+                }
+                TypeDefinitionPosition::Enum(enum_type_pos) => {
+                    let enum_type = enum_type_pos.get(schema.schema())?;
+                    let Some((first_value, _)) = enum_type.values.first() else {
+                        bail!("generate_witness_value: enum type has no values");
+                    };
+                    ast::Value::Enum(first_value.clone())
+                }
+                TypeDefinitionPosition::InputObject(input_object_type_pos) => {
+                    let object_type = input_object_type_pos.get(schema.schema())?;
+                    let fields = object_type
+                        .fields
+                        .iter()
+                        .filter_map(|(field_name, field_def)| {
+                            // We don't bother with non-mandatory fields.
+                            if !field_def.is_required() {
+                                return None;
+                            }
+
+                            let field_value = match generate_witness_value(schema, field_def) {
+                                Ok(value) => value,
+                                Err(e) => {
+                                    return Some(Err(e));
+                                }
+                            };
+                            Some(Ok((field_name.clone(), field_value)))
+                        })
+                        .collect::<Result<Vec<_>, _>>()?;
+                    ast::Value::Object(fields)
+                }
+                _ => bail!("generate_witness_value: unexpected value type"),
+            }
+        }
+        executable::Type::List(_item_type) | executable::Type::NonNullList(_item_type) => {
+            ast::Value::List(vec![])
+        }
+    };
+    Ok(Node::new(value))
+}
+
+fn display_reasons(reasons: &[Unadvanceables]) -> String {
+    let mut by_subgraph = MultiMap::new();
```

Comment:
Note that `MultiMap` internally uses `HashMap`, and we've had issues in the past with `HashMap` not providing consistent iteration order (presumably due to it being randomly seeded). Since we iterate through the `HashMap` below, we probably want to switch to e.g. `IndexMap` + `Vec` values to avoid issues with tests, or users seeing non-deterministic composition output.